csrc/lower_thread_predicate.cpp

```diff
@@ -506,6 +516,70 @@ void ThreadPredicateMap::build(Fusion* fusion) {
   for (auto expr : fusion->exprs()) {
     updateBitSet(expr);
   }
+
+  // record output tensor and its correspoinding domain merged from broadcast
+  // domain(s)
+  std::vector<std::pair<const TensorView*, IterDomain*>>
+      merged_broadcast_domains;
+  std::vector<const TensorView*> out_tv_without_broadcast_domain;
+  for (auto out_val : fusion->outputs()) {
+    auto out_tv = dynamic_cast<const TensorView*>(out_val);
+    bool tensor_with_domain_merged_from_broadcast = false;
+    auto from = out_tv->getMaybeRFactorDomain();
+    auto all_exp = DependencyCheck::getAllExprsBetween(
+        {from.begin(), from.end()},
+        {out_tv->domain()->domain().begin(), out_tv->domain()->domain().end()});
+    for (auto expr : all_exp) {
+      if (auto merge = dynamic_cast<Merge*>(expr)) {
+        auto domain = merge->out();
+        auto pt = domain->getParallelType();
+        bool from_broadcast =
+            merge->outer()->isBroadcast() || merge->inner()->isBroadcast();
+        if (isParallelTypeThread(pt) && from_broadcast) {
+          merged_broadcast_domains.emplace_back(std::make_pair(out_tv, domain));
+          tensor_with_domain_merged_from_broadcast = true;
+          break;
+        }
+      }
+    }
+    if (!tensor_with_domain_merged_from_broadcast) {
+      out_tv_without_broadcast_domain.emplace_back(out_tv);
+    }
+  }
+
+  // find correspoinding domain in non-broadcasted tensorviews with the same
+  // paralel type, set write_stride to the ratio of the
+  // extents of these two domains e.g. out_tv_ld is {I1*1}, ref_tv_ld is
+  // {I1*I2}, both are paralled by blockIdx.x since {I1*1} is merged from
+  // broadcasted domain, the write_stride should be {I1*I2} /
+  // {I1*1} = {I2} this means gmem write is only done where blockIdx.x % {I2} ==
+  // 0
```

Comment:
I think I understand what you're doing, but it seems dangerous to me. I think this is fine in your example in the test, but I don't think this would be fine in more complex situations of broadcast concretization with arbitrary splits/merges. This seems hard coded for patterns of merges then splits. Which could be fine, but we'd need a better check to ensure that's the pattern being experienced. It's a common pattern for sure, but not the only one.

Reply:
You are right. This PR only fix patterns where a domain is `isParallelTypeThread(pt) && merged_from_broadcast`. It won't touch other `broadcast concretization with arbitrary splits/merges`, actually, thread_predict can't handle these general cases. To achieve this, we need to apply the write_stride to the `index` of the tensor, which I looked and it seems to be more complex than this thread_predict approach. I can revist that more general approach.

Reply:
If you can explain to me how it only gets applied in this more isolated instance (likely I just forgot the code paths), then I'm happy to proceed and merge this PR. I just don't understand how the codebase avoids applying this rule in other situations.

Reply:
Only when a domain is `isParallelTypeThread(pt) && merged_from_broadcast.`, it will be added to the map. e.g. `iblockIdx.x103{( i0 * ( 1 * 1 ) )}`, which is parallelized by blockIdx.x and merged from broadcast domains. It won't add other domains to the map, e.g.   ` iS112{( i0 * ( 1 * 1 ) )}` becase it is not parallelized by thread/blocks, `iblockIdx.x113{( ceilDiv(( i0 * ( 1 * 1 ) ), blockDim.y) )}` becase it is not a merge from broadcast domains. 
see the following code:
```
        auto domain = merge->out();
        auto pt = domain->getParallelType();
        bool from_broadcast =
            merge->outer()->isBroadcast() || merge->inner()->isBroadcast();
        if (isParallelTypeThread(pt) && from_broadcast) {
          merged_broadcast_domains.emplace_back(std::make_pair(out_tv, domain));
          tensor_with_domain_merged_from_broadcast = true;
          break;
        }
```